server.py

```diff
@@ -27,6 +30,27 @@ def setup_logger():
             logging.getLogger(logger_name).setLevel(100)
 
 
+def validate_environment():
+    versions = {
+        'python': ('python --version', r'([0-9\.]+)', '3.6.1'),
+        'go': ('go version', r'go([0-9\.]+)', '1.11')
+    }
+    try:
+        for exe, params in versions.items():
+            cmd, pattern, min_version = params
+            failed, output = subprocess.getstatusoutput(cmd)
```

Comment:
While it probably doesn't matter here, I'm hopeful we can start restricting subprocess calls to not use shell=True.  Can you use subprocess.check_output with shell=False instead instead of getstatusoutput??   

For python you could use also do `sys.version` instead, but it breaks the pattern.

Reply:
understandable, and if we are going to put code in, better to make it something that if copied and used elsewhere in the code it wont case issues.  So I'll do a small refactor and not use the shell.
Also for the python version I thought about that but wanted to make it more expandable for future version checks if we need any.  I suppose I could key off of 'python' and split for that case and use sys.version.. hmm..

Reply:
I think it's totally reasonable to handle the python version the same way as the other versions. I just felt like throwing the option out there.